components/script/dom/gpudevice.rs

```diff
@@ -63,6 +75,74 @@ impl GPUDevice {
     }
 }
 
+impl GPUDevice {
+    fn resolve_create_buffer_mapped(
+        &self,
+        cx: SafeJSContext,
+        gpu_buffer: WebGPUBuffer,
+        array_ptr_address: usize,
+        descriptor: BufferDescriptor,
+        valid: bool,
+    ) -> Vec<JSVal> {
+        let array_ptr = array_ptr_address as *mut u8;
+
+        rooted!(in(*cx) let mut js_gpu_buffer = UndefinedValue());
+        rooted!(in(*cx) let mut js_array_buffer = ptr::null_mut::<JSObject>());
+        unsafe {
+            ArrayBuffer::create(
+                *cx,
+                CreateWith::Slice(std::slice::from_raw_parts(
+                    array_ptr as _,
+                    descriptor.size as usize,
+                )),
+                js_array_buffer.handle_mut(),
+            )
+            .is_ok()
+        };
+        let buff = GPUBuffer::new(
+            &self.global(),
+            gpu_buffer,
+            self.device,
+            GPUBufferState::Mapped,
+            descriptor.size,
+            descriptor.usage.bits(),
+            valid,
+        );
+        unsafe { buff.to_jsval(*cx, js_gpu_buffer.handle_mut()) };
+        let mut out = Vec::new();
+        out.push(js_gpu_buffer.get());
```

Comment:
This can be `ObjectValue(buff.reflector().get_jsobject().get())`, and then we don't need `js_gpu_buffer`.